patches/api/0420-Brigadier-based-command-API.patch

```diff
@@ -0,0 +1,746 @@
++     */
++    @NotNull
++    public static ArgumentType<ItemStack> itemStack() {
++        return PROVIDER.itemStack();
++    }
++
++    /**
++     * A blockstate argument which will provide rich parsing for specifying
++     * the specific block variant and then the block entity NBT if applicable.
++     *
++     * @return argument
++     */
++    @NotNull
++    public static ArgumentType<BlockState> blockState() {
++        return PROVIDER.blockState();
++    }
++
++    /**
++     * A uuid argument.
++     *
++     * @return argument
++     */
++    @NotNull
++    public static ArgumentType<UUID> uuid() {
++        return PROVIDER.uuid();
++    }
++
++    /**
++     * Represents a selector that can capture any
++     * entity.
++     *
++     * @return argument that takes one entity
++     */
++    @NotNull
++    public static ArgumentType<ArgumentResolver<Entity>> entity() {
++        return PROVIDER.entity();
++    }
++
++    /**
++     * Represents a selector that can capture multiple
++     * entities.
++     *
++     * @return argument that takes multiple entities
++     */
++    @NotNull
++    public static ArgumentType<ArgumentResolver<Collection<Entity>>> entities() {
++        return PROVIDER.entities();
++    }
++
++    /**
++     * Represents a selector that can capture a
++     * player entity.
++     *
++     * @return argument that takes one player
++     */
++    @NotNull
++    public static ArgumentType<ArgumentResolver<Player>> player() {
++        return PROVIDER.player();
++    }
++
++    /**
++     * Represents a selector that can capture multiple
++     * player entities.
++     *
++     * @return argument that takes multiple players
++     */
++    @NotNull
++    public static ArgumentType<ArgumentResolver<Collection<Player>>> players() {
++        return PROVIDER.players();
++    }
++
++    /**
++     * Represents a selector that provides list
++     * of player profiles
++     *
++     * @return player profile arguments
++     */
++    @NotNull
++    public static ArgumentType<ArgumentResolver<Collection<PlayerProfile>>> playerProfiles() {
++        return PROVIDER.playerProfiles();
++    }
++
++    /**
++     * A block position argument.
++     *
++     * @return argument
++     */
++    @NotNull
++    public static ArgumentType<ArgumentResolver<BlockPosition>> blockPos() {
++        return PROVIDER.blockPos();
++    }
++
++    /**
++     * A signed message argument.
++     * This argument can be resolved to retreieve the underlying
++     * signed message.
++     *
++     * @return argument
++     */
++    @NotNull
++    public static ArgumentType<MessageArgumentResponse> signedMessage() {
++        return PROVIDER.signedMessage();
++    }
++
++
++    /**
++     * A component argument.
++     *
++     * @return argument
++     */
++    @NotNull
++    public static ArgumentType<Component> component() {
++        return PROVIDER.component();
++    }
++
++    @ApiStatus.Internal
++    interface VanillaArgumentProvider {
```

Comment:
It might be best to move this to a top level package-private class just to not clutter the source file. Plenty of times I think people just browse source in an IDE and having extra non-api there when we already move super small interfaces to top-level package-private seems wrong.